### PR TITLE
floats: add half even rounding function

### PR DESCRIPTION
**Pull request by kortschak**
Since I was here. For some statistical things this may well be useful.

@btracey

### CODE REVIEW

File: floats_test.go [hunk 1]
```diff
@@ -1026,6 +1026,56 @@ func TestRound(t *testing.T) {
 	}
 }
 
+func TestRoundEven(t *testing.T) {
+	for _, test := range []struct {
+		x    float64
+		prec int
+		want float64
+	}{
```
Comment (btracey) [inline floats_test.go]: I don't see any tests where it rounds up to even, say 454.4455 going to 454.446

**Comment by kortschak** [comment]
Expecting correctness changes everything!

Also changed Round to use math.Pow10.

PTAL

**Comment by kortschak** [comment]
*ping*

**Comment by btracey** [comment]
lgtm

**Pull request merged**